### PR TITLE
Readd `mdoc` command

### PR DESCRIPTION
**Pull request by Darylgolden**
As the new implementation of `!mdoc` involving searching the documentation isn't quite ready, I've readded the old `!mdoc` command and renamed the new `!mdoc` to `!msearch`. This will allow us to move to using the refactor branch now.

### CODE REVIEW

**Comment by Darylgolden** [comment]
@behackl To get `!msearch` working, you may have to install some other dependencies:
 https://techoverflow.net/2018/06/05/how-to-fix-puppetteer-error-while-loading-shared-libraries-libx11-xcb-so-1-cannot-open-shared-object-file-no-such-file-or-directory/

**Comment by Darylgolden** [comment]
I've also added OpenGL rendering support, dependent on https://github.com/ManimCommunity/manim/pull/1955 and a new stable release of Manim.

**Comment by Darylgolden** [comment]
> 
> 
> This pull request **introduces 1 alert** when merging [136164e](https://github.com/ManimCommunity/DiscordManimator/commit/136164e07abe45c2bad31092ffb578ee2ed9ba16) into [db46812](https://github.com/ManimCommunity/DiscordManimator/commit/db4681295d78cbb66fc4ab4413a3763b2a63c33a) - [view on LGTM.com](https://lgtm.com/projects/g/ManimCommunity/DiscordManimator/rev/pr-3f850fe47ad48d2af1c3697d984a34d291c06306)
> 
> **new alerts:**
> 
>     * 1 for Unused import

I'm sorry Mr. Bot 😔

**Review by behackl — APPROVED** [review]
`requests-html` says that it *only* supports Python 3.6, is this actually the case?

I'm a bit unsure about using `requests-html` over a more common combination (like `requests` for fetching the site + `beautifulsoup` for parsing the html), but in terms of this PR this seems fine to me.

I haven't tried running this on the actual server where Manimator runs, but can do that soon. We can still merge this and deal with any potential problems afterwards.

**Comment by Darylgolden** [comment]
> `requests-html` says that it _only_ supports Python 3.6, is this actually the case?

Don't think so, it's running fine with Python 3.8.5.



> I'm a bit unsure about using `requests-html` over a more common combination (like `requests` for fetching the site + `beautifulsoup` for parsing the html), but in terms of this PR this seems fine to me.

I suppose you'll have to ask @infiniteregrets about that, I merely renamed the file and command. Perhaps it's worth rewriting as you mentioned.

File: cogs/manimate.py [hunk 1]
```diff
@@ -33,16 +33,20 @@ def construct_reply(arg):
                 "--save_last_frame",
                 "-t",
                 "--transparent",
+                "--renderer=opengl"
             ]
+
             if not all([flag in allowed_flags for flag in cli_flags]):
                 reply_args = {
```
[the remark this answers is not in the document]
Comment (Darylgolden) [reply cogs/manimate.py:40]: Good point, I've added it.

**Comment by infiniteregrets** [comment]
Try doing a requests.get() on the manim website with a search query in the URL and you'll the page doesn't load completely. requests_html helps with that, otherwise we would have to use something like selenium which is a lot of hassle. 

I worked on this a while ago, let me know if this is not the case anymore and we'll what we can do

**Comment by behackl** [comment]
> Try doing a requests.get() on the manim website with a search query in the URL and you'll the page doesn't load completely. requests_html helps with that, otherwise we would have to use something like selenium which is a lot of hassle.
> 
> I worked on this a while ago, let me know if this is not the case anymore and we'll what we can do

Can't we query the API directly?

**Comment by infiniteregrets** [comment]
There is no API @behackl

**Comment by behackl** [comment]
> There is no API @behackl

I don't think that [this is entirely true](https://docs.readthedocs.io/en/stable/server-side-search.html).

```py
>>> import requests
>>> req = requests.get("https://docs.manim.community/_/api/v2/search/?q=table&project=manimce&version=stable&language=en")
>>> res = req.json()
>>> for queryres in res["results"]:
...     print(queryres["title"], queryres["path"])
... 
Table /en/stable/reference/manim.mobject.table.Table.html
table /en/stable/reference/manim.mobject.table.html
MathTable /en/stable/reference/manim.mobject.table.MathTable.html
IntegerTable /en/stable/reference/manim.mobject.table.IntegerTable.html
MobjectTable /en/stable/reference/manim.mobject.table.MobjectTable.html
DecimalTable /en/stable/reference/manim.mobject.table.DecimalTable.html
manim.mobject.table /en/stable/_modules/manim/mobject/table.html
v0.9.0 /en/stable/changelog/0.9.0-changelog.html
Mobjects /en/stable/reference_index/mobjects.html
manim.mobject.svg.code_mobject /en/stable/_modules/manim/mobject/svg/code_mobject.html
```

**Comment by infiniteregrets** [comment]
Um I am not sure about this. Are you able to parse the data into the current format

**Comment by behackl** [comment]
> Um I am not sure about this. Are you able to parse the data into the current format

See #27 for an attempt at it. Let's merge this PR and continue discussion there.

**Pull request merged**